service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -262,18 +262,33 @@ public EhrExtractStatus updateEhrExtractStatusAccessDocumentDocumentReferences(
         return ehrExtractStatus;
     }
 
-    public void updateEhrExtractStatusAcknowledgement(SendAcknowledgementTaskDefinition sendAcknowledgementTaskDefinition,
-        String positiveAckMessageId) {
-        Query query = createQueryForConversationId(sendAcknowledgementTaskDefinition.getConversationId());
+    public void updateEhrExtractStatusPositiveAcknowledgement(SendAcknowledgementTaskDefinition sendAcknowledgementTaskDefinition,
+        String ackMessageId) {
+        Update update = createUpdateWithUpdatedAt();
+        updateEhrExtractStatusAcknowledgement2(sendAcknowledgementTaskDefinition, ackMessageId, update);
+    }
+
+    public void updateEhrExtractStatusNegativeAcknowledgement(
+        SendNegativeAcknowledgementTaskDefinition sendNegativeAcknowledgementTaskDefinition,
+        String ackMessageId) {
 
         Update update = createUpdateWithUpdatedAt();
+        update.set(ACK_REASON_CODE_PATH, sendNegativeAcknowledgementTaskDefinition.getReasonCode());
+        update.set(ACK_DETAIL_CODE_PATH, sendNegativeAcknowledgementTaskDefinition.getDetail());
+
+        updateEhrExtractStatusAcknowledgement2(sendNegativeAcknowledgementTaskDefinition, ackMessageId, update);
+    }
+
+    private void updateEhrExtractStatusAcknowledgement2(
```

Comment:
Rename to something more descriptive like

First parts could be, add positive/negativeToUpdate, then call updateEhrExtractStatusAcknowledgement?

Reply:
yes, I missed it, sorry